src/classes/SRVC_Contact_PrimaryAffiliations.cls

```diff
@@ -0,0 +1,306 @@
+                         try {
+                             afflToCreate = new Affiliation__c(Contact__c = newContact.ID, Account__c = (Id)newContact.get(eachModel.getContactFieldApiName()), Primary__c = true);
+                         } catch (System.TypeException typeExc) {
+                             newContact.addError(typeExc.getMessage());
+                             throw new ERR_ExceptionHandler.AffAccountRecordTypeMappingsException(Label.afflAccoutMappingError);
+                         } catch (System.StringException stringExc) {
+                             newContact.addError(stringExc.getMessage());
+                             throw new ERR_ExceptionHandler.AffAccountRecordTypeMappingsException(Label.afflAccoutMappingError);
+                         }
+                         affiliationsToCreateForEachContactList.add(afflToCreate);
+                }
+                if (oldContact.get(eachModel.getContactFieldApiName()) != null && newContact.get(eachModel.getContactFieldApiName()) == null) {
+                     accountIds.add((Id)oldContact.get(eachModel.getContactFieldApiName()));
+                }
+            }
+        } catch (ERR_ExceptionHandler.AffAccountRecordTypeMappingsException error) {
+           AffiliationsToProcess AfflsToProcess = new AffiliationsToProcess();
+           return AfflsToProcess;
+        }
+
+        AffiliationsToProcess AfflsToProcess = new AffiliationsToProcess(affiliationsToCreateForEachContactList, accountIds);
+        return AfflsToProcess;
+    }
+
+    /************************************************************************
+    * @description Returns a list of Affiliations to update for each Contact
+    * @Param newContact The contact to update Affiliations for
+    * @Param newContact The contact to before update
+    * @Param afflMappingsModelsList List of Primary Affiliations Models
+    ************************************************************************/
+    @TestVisible 
+    Private AffiliationsToProcess getAffiliationsToUpdateForContact(Contact newContact, Contact oldContact, List<MODL_PrimaryAfflMapper> afflMappingsModelsList) {
+
+        List<Affiliation__c> affiliationsToCreateForEachContactList = new List<Affiliation__c>();
+        Set<Id> accountIds = new Set<Id>();
+
+        for (MODL_PrimaryAfflMapper eachModel: afflMappingsModelsList) {
+
+            if (newContact.get(eachModel.getContactFieldApiName()) == oldContact.get(eachModel.getContactFieldApiName())) {
+                continue;
+            }
+
+            if (oldContact.get(eachModel.getContactFieldApiName()) != null && newContact.get(eachModel.getContactFieldApiName()) == null) {
+                accountIds.add((Id)oldContact.get(eachModel.getContactFieldApiName()));
+            }
+        }
+
+        AffiliationsToProcess AfflsToProcess = new AffiliationsToProcess(affiliationsToCreateForEachContactList, accountIds);
+        return AfflsToProcess;
```

Comment:
@MatthewBlanski wanted to check with you on this one, 'affiliationsToCreateForEachContactList' is empty in this method. Should we change this?